File: TicTacToe.Api/Models/GameState.cs
using TicTacToe.Api.Enums;

namespace TicTacToe.Api.Models;

public class GameState
{
    public Guid Id { get; set; }

  public string[][] Board { get; set; } = [];

    public Player CurrentPlayer { get; set; }

    public GameMode Mode { get; set; }

    public GameStatus Status { get; set; }

    public Player? Winner { get; set; }

    public List<int> WinningCells { get; set; } = new();

    public List<Move> MoveHistory { get; set; } = new();
}

File: tic-tac-toe-ui/src/app/models/game-state.ts
export interface Move {
  moveNumber: number;
  player: number;
  row: number;
  column: number;
}

export interface GameState {
  id: string;
  board: string[][];
  currentPlayer: number;
  mode: number;
  status: number;
  winner?: number;
  winningCells: number[];
  moveHistory: Move[];
}